src/Microsoft.TemplateEngine.Abstractions/IPostAction.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Abstractions
+{
+    public interface IPostAction
+    {
+        int Order { get; }
+
+        string Name { get; }
+
+        IReadOnlyList<IPostActionOperation> Operations { get; }
+
+        IReadOnlyList<IPostActionOperation> AlternateOperations { get; }
```

Comment:
I think these are intended to just be user facing instructions, we should rename/retype as such. Perhaps following up with Damian for clarification on what should happen if we didn't understand the command family (`name` I'm guessing is the equivalent to the family of commands).

I'm also a little unclear on `CommandText`, does it give the flexibility for callers to understand how to do parameter substitutions or whatnot if required? I suppose they could run through the processor to replace tokens and whatnot, but giving some sort of a structured payload (maybe just key value pairs) instead and leaving it up to the thing that interprets the command family in the caller to appropriately structure the command line or determine the component in the caller (maybe it invokes something in VS/XS/etc. to take an action instead of shelling out).

My imagination on how this would be populated (without changing away from CommandText):
Order => 0
Name => NPM
Operations => [ { CommandText => "npm install" } ]
ManualInstructions (or something like that) => [ "Run 'npm install', if you don't have npm, you can get it from (place that one gets npm from)" ]